pcdsdevices/ipm.py

```diff
@@ -49,15 +96,276 @@ def removed(self):
+    ypos = Cpt(EpicsSignal, ':YPOS', kind='normal')
+    evr_channel = Cpt(Trigger, ':TRIG:TRIG0', kind='normal')
+    ch0 = FCpt(Wave8Channel, '{self.prefix}', channel_index=0, kind='normal')
+    ch1 = FCpt(Wave8Channel, '{self.prefix}', channel_index=1, kind='normal')
+    ch2 = FCpt(Wave8Channel, '{self.prefix}', channel_index=2, kind='normal')
+    ch3 = FCpt(Wave8Channel, '{self.prefix}', channel_index=3, kind='normal')
+    ch4 = FCpt(Wave8Channel, '{self.prefix}', channel_index=4, kind='normal')
+    ch5 = FCpt(Wave8Channel, '{self.prefix}', channel_index=5, kind='normal')
+    ch6 = FCpt(Wave8Channel, '{self.prefix}', channel_index=6, kind='normal')
+    ch7 = FCpt(Wave8Channel, '{self.prefix}', channel_index=7, kind='normal')
+    ch8 = FCpt(Wave8Channel, '{self.prefix}', channel_index=8, kind='normal')
+    ch9 = FCpt(Wave8Channel, '{self.prefix}', channel_index=9, kind='normal')
+    ch10 = FCpt(Wave8Channel, '{self.prefix}', channel_index=10, kind='normal')
+    ch11 = FCpt(Wave8Channel, '{self.prefix}', channel_index=11, kind='normal')
+    ch12 = FCpt(Wave8Channel, '{self.prefix}', channel_index=12, kind='normal')
+    ch13 = FCpt(Wave8Channel, '{self.prefix}', channel_index=13, kind='normal')
+    ch14 = FCpt(Wave8Channel, '{self.prefix}', channel_index=14, kind='normal')
+    ch15 = FCpt(Wave8Channel, '{self.prefix}', channel_index=15, kind='normal')
+
+    def __init__(self, prefix, *, name, prefix_ioc=None, **kwargs):
+        if not prefix_ioc:
+            self._prefix_ioc = 'IOC:%s' % prefix
+        else:
+            self._prefix_ioc = prefix_ioc
+        super().__init__(prefix, name=name, **kwargs)
+
+    def screen(self):
+        """
+        Function to call the (pyQT) screen for a wave8 box
+        """
+        executable = '/reg/g/pcds/pyps/apps/wave8/latest/wave8'
+        if shutil.which(executable) is None:
+            print('%s is not on path, we cannot start the screen' % executable)
+            return
+        os.system('%s --base %s --ioc %s --evr %s &' %
+                  (executable, self.prefix, self._prefix_ioc,
+                   self.prefix+':TRIG'))
+
+
+class IPM_IPIMB(IPMMotion):
+    """
+%s
+
+    has a member ipimb which represents the ipimb box used for readout
+    """
+    __doc__ = __doc__ % (IPM_base) + basic_positioner_init
+
+    ipimb = FCpt(IPIMB, '{self.prefix_ipimb}', prefix_ioc='{self.prefix_ioc}')
+
+    def __init__(self, prefix, *, name, **kwargs):
+        self.prefix_ipimb = kwargs.pop('prefix_ipimb')
+        self.prefix_ioc = kwargs.pop('prefix_ioc', None)
+        super().__init__(prefix, name=name, **kwargs)
+
+    def isum(self):
+        return self.ipimb.isum.get()
+
+    def xpos(self):
+        return self.ipimb.xpos.get()
+
+    def ypos(self):
+        return self.ipimb.ypos.get()
+
+    def channel(self, i=0):
+        # thrown an exception for an invalid channel?
+        # if ( i >= 4 ): return None
+        return getattr(self.ipimb, 'ch%i').amplitude.get()
```

Comment:
Definitely should throw an exception here as per the first comment. The exception that will be thrown otherwise is not going to be helpful. Note that you should also catch negative channel inputs unlike the second comment.